### PR TITLE
Add a way of customising the comparison used in assert_roundtrip_tree

### PR DESCRIPTION
**Pull request by Cadair**
Happy new year!

### CODE REVIEW

File: asdf/tests/helpers.py [hunk 1]
```diff
@@ -113,8 +118,8 @@ def recurse(old, new):
 
 
 def assert_roundtrip_tree(
-        tree, tmpdir, asdf_check_func=None, raw_yaml_check_func=None,
-        write_options={}, extensions=None):
+        tree, tmpdir, asdf_check_func=None, tree_match_func='assert_equal',
```
Comment (Cadair) [inline asdf/tests/helpers.py]: I feel that `asdf_check_func` and `tree_match_func` are a little redundant here. Do you think it makes sense to remove `asdf_check_func`?

Comment (drdavella) [reply asdf/tests/helpers.py]: It does seem redundant. We'll have to go through and update all of the test cases that currently make use of it.

**Comment by Cadair** [comment]
@drdavella I tried to remove the `asdf_check_func` argument, but the key difference is that it takes a asdf file object not the trees. So I think we need both. I did go through and change all uses of `assert_rountrip_tree` to use kwargs. I can remove the `*,` if you don't want to enforce that, but I thought it was a better idea in the code.

**Comment by drdavella** [comment]
Thanks @Cadair!

**Pull request merged**